ts/input/tex/Configuration.ts

```diff
@@ -278,3 +233,169 @@ export namespace ConfigurationHandler {
+   */
+  protected configurations: PrioritizedList<Configuration> = new PrioritizedList();
+
+  /**
+   * The subhandlers for this configuration.
+   * @type {SubHandlers}
+   */
+  public handlers: SubHandlers = new SubHandlers();
+
+  /**
+   * The collated stack items.
+   * @type {StackItemConfig}
+   */
+  public items: StackItemConfig = {};
+
+  /**
+   * The collated tag configurations.
+   * @type {TagsConfig}
+   */
+  public tags: TagsConfig = {};
+
+  /**
+   * The collated options.
+   * @type {OptionList}
+   */
+  public options: OptionList = {};
+
+  /**
+   * The collated node creators.
+   * @type {{[key: string]: any}}
+   */
+  public nodes: {[key: string]: any}  = {};
+
+
+  /**
+   * @constructor
+   */
+  constructor(packages: (string | [string, number])[]) {
+    for (const pkg of packages.slice().reverse()) {
+      this.addPackage(pkg);
+    }
+    for (let {item: config, priority: priority} of this.configurations) {
+      this.append(config, priority);
+    }
+  }
+
+  /**
+   * Initmethod for the configuration;
+   *
+   * @param {Configuration} configuration   The configuration where this one is being initialized
+   */
+  public init() {
+    this.initMethod.execute(this);
+  }
+
+  /**
+   * Init method for when the jax is ready
+   *
+   * @param {Configuration} configuration   The configuration where this one is being initialized
+   * @param {TeX} jax                       The TeX jax for this configuration
+   */
+  public config(jax: TeX<any, any, any>) {
+    this.configMethod.execute(this, jax);
+    for (const config of this.configurations) {
+      this.addFilters(jax, config.item);
+    }
+  }
+
+  /**
+   * Retrieves and adds configuration for a pacakge with priority.
+   * @param {(string | [string, number]} pkg Package with priority.
+   */
+  public addPackage(pkg: (string | [string, number])) {
+    const name = typeof pkg === 'string' ? pkg : pkg[0];
+    let conf = ConfigurationHandler.get(name);
+    if (conf) {
+      this.configurations.add(
+        conf, typeof pkg === 'string' ? conf.priority : pkg[1]);
+    }
+  }
+
+  /**
+   * Adds a configuration after the input jax is created.  (Used by \require.)
+   * Sets items, nodes and runs configuration method explicitly.
+   *
+   * @param {Configuration} config   The configuration to be registered in this one
+   * @param {TeX} jax                The TeX jax where it is being registered
+   * @param {OptionList=} options    The options for the configuration.
+   */
+  public add(config: Configuration, jax: TeX<any, any, any>, options: OptionList = {}) {
+    this.append(config);
+    this.configurations.add(config, config.priority);
+    this.init();
+    const parser = jax.parseOptions;
+    parser.nodeFactory.setCreators(config.nodes);
+    for (const kind of Object.keys(config.items)) {
+      parser.itemFactory.setNodeClass(kind, config.items[kind]);
+    }
+    TagsFactory.addTags(config.tags);
+    userOptions(parser.options, options);
```

Comment:
You need to add

```
    defaultOptions(parser.options, config.options);
```

before this line otherwise packages that have configuration options (like `amscd`, `color`, etc.) will not get their default configurations, and will throw an error trying to set their user-supplied options at this line.  That can be tested using

```
node -r esm
require('./components/src/node-main/node-main.js').init({loader: {load: ['input/tex', '[tex]/require']}}).then(() => MathJax.tex2mmlPromise('\\require{color} {\\color{red} x}')).then((x) => console.log(x)).catch((e) => console.log(e.message))
```

for example.

Reply:
Thanks for spotting this.